### DISCUSSION TITLE
How to tell if a timezone has DST?

### DISCUSSION THREAD

**Discussion by PaulAtST** [moment/luxon#1172] [Q&A] [answered]
After creating a date with a set timezone, I am trying to determine if the timezone supports daylights savings time or has a constant offset year round. 

I was under the impression from the documentation I could use `isUniversal` but I am seeing it report false for timezones that do and do not support daylights savings time. 

What's the best way to check if a time zones offset changes throughout the year?

**Comment by icambron** [comment, marked as answer]
Unfortunately, the only way is to probe some dates and see if they have different offsets. Luxon users a similar trick to implement `isInDST()`

Luxon doesn't actually have access to the time zone rules, so it doesn't have a way to tell you that from the zone itself. `isUniversal`, as you saw, can only know if it can't have a DST because the zone declares itself that way, so it works with fixed-offset zones that Luxon supports directly. For an IANA-specified zone, it might or might not have a DST, and Luxon will get the times right regardless, but it's unaware of the DST as a standalone fact.

**Comment by PaulAtST** [comment]
@icambron Thank you for the info. I might be able to do something similar for my use case. 

It would be nice if Intl API would eventually support labeling which time zones support DST, and if you could equally get the start and end dates. Do you happen know if there is a way to make Intl API suggestions?

**Reply by icambron** [reply]
My guess is that the Intl API would be unlikely to add that, mostly because Luxon is abusing it horribly; its intended purpose isn't to support datetime libraries, just format stuff. Another reason is that the upcoming Temporal API lets you check this directly: https://tc39.es/proposal-temporal/docs/timezone.html#getNextTransition

**Comment by ajmas** [comment]
One hack could be to take a January 1st and and December 1st and check their UTC offset, and then see if their is a delta?

```js
function usesDST () {
  const timeNow = DateTime.now();
  const time1 = timeNow.set({ day: 1, month: 6, hour: 12, minute: 0, seconds: 0 });
  const time2 = timeNow.set({ day: 1, month: 12, hour: 12, minute: 0, seconds: 0 });
  
  return time1.toUTC().get('hour') !== time2.toUTC().get('hour');
}
```

Then extend the logic to see whether your current time is using the same offset as the DST / non-DST variant. 

BTW The above code is implemented the way it is to factor in an possible DST logic changes for a given year, such as if DST is no longer used (as is often proposed in certain countries). Also not calling `Datetime.now()` twice to avoid possible hour change between calls (maybe a low probability, but why take a chance?).

As for the Temporal API, it still hasn't really gained any adoption at this pont
